src/System.Net.Http/src/System/Net/Http/HttpClientHandler.Net46.cs

```diff
@@ -0,0 +1,1199 @@
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Diagnostics.Contracts;
+using System.Globalization;
+using System.IO;
+using System.Net.Http.Headers;
+using System.Net.Security;
+using System.Runtime.CompilerServices;
+using System.Security;
+using System.Security.Authentication;
+using System.Security.Cryptography.X509Certificates;
+using System.Security.Principal;
+using System.Security.Permissions;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http
+{
+    public class HttpClientHandler : HttpMessageHandler
+    {
+        #region Fields
+
+        private static readonly Action<object> s_onCancel = OnCancel;
+
+        private readonly Action<object> _startRequest;
+        private readonly AsyncCallback _getRequestStreamCallback;
+        private readonly AsyncCallback _getResponseCallback;
+
+        private volatile bool _operationStarted;
+        private volatile bool _disposed;
+
+        private long _maxRequestContentBufferSize;
+        private int _maxResponseHeadersLength;
+        private CookieContainer _cookieContainer;
+        private bool _useCookies;
+        private DecompressionMethods _automaticDecompression;
+        private IWebProxy _proxy;
+        private bool _useProxy;
+        private ICredentials _defaultProxyCredentials;
+        private bool _preAuthenticate;
+        private bool _useDefaultCredentials;
+        private ICredentials _credentials;
+        private bool _allowAutoRedirect;
+        private int _maxAutomaticRedirections;
+        private string _connectionGroupName;
+        private ClientCertificateOption _clientCertOptions;
+        private X509Certificate2Collection _clientCertificates;
+        private IDictionary<String, Object> _properties;
+        private int _maxConnectionsPerServer;
+        private Func<
+            HttpRequestMessage,
+            X509Certificate2,
+            X509Chain,
+            SslPolicyErrors,
+            bool> _serverCertificateCustomValidationCallback;
+
+        #endregion Fields
+
+        #region Properties
+
+        public bool AllowAutoRedirect
+        {
+            get
+            {
+                return _allowAutoRedirect;
+            }
+
+            set
+            {
+                CheckDisposedOrStarted();
+                _allowAutoRedirect = value;
+            }
+        }
+
+        public DecompressionMethods AutomaticDecompression
+        {
+            get
+            {
+                return _automaticDecompression;
+            }
+
+            set
+            {
+                CheckDisposedOrStarted();
+                _automaticDecompression = value;
+            }
+        }
+
+        public bool CheckCertificateRevocationList
+        {
+            // New property not in .NET Framework HttpClientHandler.
+            get
+            {
+                throw new PlatformNotSupportedException();
```

Comment:
I've reviewed possible designs using existing ServicePointManager implementation. There is no perfect solution.  Any implementation of these properties would end up modifying a ServicePointManager property which is basically global. That would introduce race conditions on the actual value used during an http request. 

If we're comfortable with that uncertainty in the design, then we could do changes for that. and I could create a tracking issue for it. However, the current thinking is that it is better to just throw.